teamcity: failed tests on master: test (proposer evaluated kv)/TestStoreRangeGossipOnSplits, testrace (proposer evaluated kv)/TestLogic, testrace (proposer evaluated kv)/TestLogic/distsql, testrace (proposer evaluated kv)/TestLogic/distsql/pipelining

The following tests appear to have failed:

[#205106](https://teamcity.cockroachdb.com/viewLog.html?buildId=205106):

```
--- FAIL: test (proposer evaluated kv)/TestStoreRangeGossipOnSplits (0.000s)
Test ended in panic.

------- Stdout: -------
I170403 16:01:50.407050 39045 gossip/gossip.go:293  [n1] NodeDescriptor set to node_id:1 address:<network_field:"" address_field:"" > attrs:<> locality:<> 
I170403 16:01:50.408153 39045 storage/store.go:1329  [n1,s1]: failed initial metrics computation: [n1,s1]: system config not yet available
panic: test timed out after 3m0s

goroutine 39338 [running]:
testing.startAlarm.func1()
	/usr/local/go/src/testing/testing.go:1023 +0xf9
created by time.goFunc
	/usr/local/go/src/time/sleep.go:170 +0x44

goroutine 1 [chan receive, 2 minutes]:
testing.(*T).Run(0xc4201bc4e0, 0x1d353c8, 0x1c, 0x1d91248, 0xc420497a01)
	/usr/local/go/src/testing/testing.go:698 +0x2f4
testing.runTests.func1(0xc4201bc4e0)
	/usr/local/go/src/testing/testing.go:882 +0x67
testing.tRunner(0xc4201bc4e0, 0xc420497b78)
	/usr/local/go/src/testing/testing.go:657 +0x96
testing.runTests(0xc42000c460, 0x2986860, 0x19b, 0x19b, 0x695235)
	/usr/local/go/src/testing/testing.go:888 +0x2c1
testing.(*M).Run(0xc42048df20, 0xc420497d70)
	/usr/local/go/src/testing/testing.go:822 +0xfc
github.com/cockroachdb/cockroach/pkg/storage_test.TestMain(0xc420497f20)
	/go/src/github.com/cockroachdb/cockroach/pkg/storage/main_test.go:59 +0x20a
main.main()
	github.com/cockroachdb/cockroach/pkg/storage/_test/_testmain.go:872 +0xf7

goroutine 17 [syscall, 3 minutes, locked to thread]:
runtime.goexit()
	/usr/local/go/src/runtime/asm_amd64.s:2197 +0x1

goroutine 5 [chan receive]:
github.com/cockroachdb/cockroach/pkg/util/log.(*loggingT).flushDaemon(0x2ddaae0)
	/go/src/github.com/cockroachdb/cockroach/pkg/util/log/clog.go:1004 +0x77
created by github.com/cockroachdb/cockroach/pkg/util/log.init.1
	/go/src/github.com/cockroachdb/cockroach/pkg/util/log/clog.go:583 +0xd2

goroutine 6 [chan receive, 3 minutes]:
github.com/cockroachdb/cockroach/pkg/util/log.(*loggingT).gcDaemon(0x2ddaae0)
	/go/src/github.com/cockroachdb/cockroach/pkg/util/log/clog.go:1027 +0x5b
created by github.com/cockroachdb/cockroach/pkg/util/log.init.1
	/go/src/github.com/cockroachdb/cockroach/pkg/util/log/clog.go:584 +0xf6

goroutine 39212 [select, 2 minutes]:
github.com/cockroachdb/cockroach/pkg/storage.(*baseQueue).processLoop.func1()
	/go/src/github.com/cockroachdb/cockroach/pkg/storage/queue.go:490 +0x39e
github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).RunWorker.func1(0xc421324000, 0xc42015e060)
	/go/src/github.com/cockroachdb/cockroach/pkg/util/stop/stopper.go:210 +0x7d
created by github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).RunWorker
	/go/src/github.com/cockroachdb/cockroach/pkg/util/stop/stopper.go:211 +0x66

goroutine 39046 [chan receive, 2 minutes]:
github.com/cockroachdb/cockroach/pkg/storage/engine.(*RocksDB).open.func2(0xc420f9a000)
	/go/src/github.com/cockroachdb/cockroach/pkg/storage/engine/rocksdb.go:423 +0x44
created by github.com/cockroachdb/cockroach/pkg/storage/engine.(*RocksDB).open
	/go/src/github.com/cockroachdb/cockroach/pkg/storage/engine/rocksdb.go:424 +0x4c1

goroutine 39053 [semacquire]:
sync.runtime_notifyListWait(0xc4204be610, 0xc40000056a)
	/usr/local/go/src/runtime/sema.go:297 +0x10b
sync.(*Cond).Wait(0xc4204be600)
	/usr/local/go/src/sync/cond.go:57 +0x89
github.com/cockroachdb/cockroach/pkg/storage.(*raftScheduler).worker(0xc420370e10, 0xc421324000)
	/go/src/github.com/cockroachdb/cockroach/pkg/storage/scheduler.go:196 +0xac
github.com/cockroachdb/cockroach/pkg/storage.(*raftScheduler).Start.func2()
	/go/src/github.com/cockroachdb/cockroach/pkg/storage/scheduler.go:165 +0x33
github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).RunWorker.func1(0xc421324000, 0xc4213feec0)
	/go/src/github.com/cockroachdb/cockroach/pkg/util/stop/stopper.go:210 +0x7d
created by github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).RunWorker
	/go/src/github.com/cockroachdb/cockroach/pkg/util/stop/stopper.go:211 +0x66

goroutine 39371 [semacquire, 2 minutes]:
sync.runtime_Semacquire(0xc420370e94)
	/usr/local/go/src/runtime/sema.go:47 +0x34
sync.(*WaitGroup).Wait(0xc420370e88)
	/usr/local/go/src/sync/waitgroup.go:131 +0x7a
github.com/cockroachdb/cockroach/pkg/storage.(*raftScheduler).Wait(0xc420370e10)
	/go/src/github.com/cockroachdb/cockroach/pkg/storage/scheduler.go:171 +0x31
github.com/cockroachdb/cockroach/pkg/storage.(*raftScheduler).Wait-fm()
	/go/src/github.com/cockroachdb/cockroach/pkg/storage/store.go:3482 +0x2a
github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).RunWorker.func1(0xc421324000, 0xc4207adde0)
	/go/src/github.com/cockroachdb/cockroach/pkg/util/stop/stopper.go:210 +0x7d
created by github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).RunWorker
	/go/src/github.com/cockroachdb/cockroach/pkg/util/stop/stopper.go:211 +0x66

goroutine 39375 [select, 2 minutes]:
github.com/cockroachdb/cockroach/pkg/storage.(*Store).Start.func3()
	/go/src/github.com/cockroachdb/cockroach/pkg/storage/store.go:1299 +0x165
github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).RunWorker.func1(0xc421324000, 0xc4213ff0e0)
	/go/src/github.com/cockroachdb/cockroach/pkg/util/stop/stopper.go:210 +0x7d
created by github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).RunWorker
	/go/src/github.com/cockroachdb/cockroach/pkg/util/stop/stopper.go:211 +0x66

goroutine 39051 [semacquire]:
sync.runtime_notifyListWait(0xc4204be610, 0xc400000568)
	/usr/local/go/src/runtime/sema.go:297 +0x10b
sync.(*Cond).Wait(0xc4204be600)
	/usr/local/go/src/sync/cond.go:57 +0x89
github.com/cockroachdb/cockroach/pkg/storage.(*raftScheduler).worker(0xc420370e10, 0xc421324000)
	/go/src/github.com/cockroachdb/cockroach/pkg/storage/scheduler.go:196 +0xac
github.com/cockroachdb/cockroach/pkg/storage.(*raftScheduler).Start.func2()
	/go/src/github.com/cockroachdb/cockroach/pkg/storage/scheduler.go:165 +0x33
github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).RunWorker.func1(0xc421324000, 0xc4213fee80)
	/go/src/github.com/cockroachdb/cockroach/pkg/util/stop/stopper.go:210 +0x7d
created by github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).RunWorker
	/go/src/github.com/cockroachdb/cockroach/pkg/util/stop/stopper.go:211 +0x66

goroutine 39366 [semacquire]:
sync.runtime_notifyListWait(0xc4204be610, 0xc40000056d)
	/usr/local/go/src/runtime/sema.go:297 +0x10b
sync.(*Cond).Wait(0xc4204be600)
	/usr/local/go/src/sync/cond.go:57 +0x89
github.com/cockroachdb/cockroach/pkg/storage.(*raftScheduler).worker(0xc420370e10, 0xc421324000)
	/go/src/github.com/cockroachdb/cockroach/pkg/storage/scheduler.go:196 +0xac
github.com/cockroachdb/cockroach/pkg/storage.(*raftScheduler).Start.func2()
	/go/src/github.com/cockroachdb/cockroach/pkg/storage/scheduler.go:165 +0x33
github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).RunWorker.func1(0xc421324000, 0xc4213fefe0)
	/go/src/github.com/cockroachdb/cockroach/pkg/util/stop/stopper.go:210 +0x7d
created by github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).RunWorker
	/go/src/github.com/cockroachdb/cockroach/pkg/util/stop/stopper.go:211 +0x66

goroutine 39052 [semacquire]:
sync.runtime_notifyListWait(0xc4204be610, 0xc400000577)
	/usr/local/go/src/runtime/sema.go:297 +0x10b
sync.(*Cond).Wait(0xc4204be600)
	/usr/local/go/src/sync/cond.go:57 +0x89
github.com/cockroachdb/cockroach/pkg/storage.(*raftScheduler).worker(0xc420370e10, 0xc421324000)
	/go/src/github.com/cockroachdb/cockroach/pkg/storage/scheduler.go:196 +0xac
github.com/cockroachdb/cockroach/pkg/storage.(*raftScheduler).Start.func2()
	/go/src/github.com/cockroachdb/cockroach/pkg/storage/scheduler.go:165 +0x33
github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).RunWorker.func1(0xc421324000, 0xc4213feea0)
	/go/src/github.com/cockroachdb/cockroach/pkg/util/stop/stopper.go:210 +0x7d
created by github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).RunWorker
	/go/src/github.com/cockroachdb/cockroach/pkg/util/stop/stopper.go:211 +0x66

goroutine 39055 [semacquire]:
sync.runtime_notifyListWait(0xc4204be610, 0xc400000576)
	/usr/local/go/src/runtime/sema.go:297 +0x10b
sync.(*Cond).Wait(0xc4204be600)
	/usr/local/go/src/sync/cond.go:57 +0x89
github.com/cockroachdb/cockroach/pkg/storage.(*raftScheduler).worker(0xc420370e10, 0xc421324000)
	/go/src/github.com/cockroachdb/cockroach/pkg/storage/scheduler.go:196 +0xac
github.com/cockroachdb/cockroach/pkg/storage.(*raftScheduler).Start.func2()
	/go/src/github.com/cockroachdb/cockroach/pkg/storage/scheduler.go:165 +0x33
github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).RunWorker.func1(0xc421324000, 0xc4213fef00)
	/go/src/github.com/cockroachdb/cockroach/pkg/util/stop/stopper.go:210 +0x7d
created by github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).RunWorker
	/go/src/github.com/cockroachdb/cockroach/pkg/util/stop/stopper.go:211 +0x66

goroutine 39209 [select, 2 minutes]:
github.com/cockroachdb/cockroach/pkg/storage.(*baseQueue).processLoop.func1()
	/go/src/github.com/cockroachdb/cockroach/pkg/storage/queue.go:490 +0x39e
github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).RunWorker.func1(0xc421324000, 0xc42015e000)
	/go/src/github.com/cockroachdb/cockroach/pkg/util/stop/stopper.go:210 +0x7d
created by github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).RunWorker
	/go/src/github.com/cockroachdb/cockroach/pkg/util/stop/stopper.go:211 +0x66

goroutine 39364 [semacquire]:
sync.runtime_notifyListWait(0xc4204be610, 0xc400000575)
	/usr/local/go/src/runtime/sema.go:297 +0x10b
sync.(*Cond).Wait(0xc4204be600)
	/usr/local/go/src/sync/cond.go:57 +0x89
github.com/cockroachdb/cockroach/pkg/storage.(*raftScheduler).worker(0xc420370e10, 0xc421324000)
	/go/src/github.com/cockroachdb/cockroach/pkg/storage/scheduler.go:196 +0xac
github.com/cockroachdb/cockroach/pkg/storage.(*raftScheduler).Start.func2()
	/go/src/github.com/cockroachdb/cockroach/pkg/storage/scheduler.go:165 +0x33
github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).RunWorker.func1(0xc421324000, 0xc4213fefa0)
	/go/src/github.com/cockroachdb/cockroach/pkg/util/stop/stopper.go:210 +0x7d
created by github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).RunWorker
	/go/src/github.com/cockroachdb/cockroach/pkg/util/stop/stopper.go:211 +0x66

goroutine 39372 [select]:
github.com/cockroachdb/cockroach/pkg/storage.(*Store).raftTickLoop.func1()
	/go/src/github.com/cockroachdb/cockroach/pkg/storage/store.go:3499 +0x337
github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).RunWorker.func1(0xc421324000, 0xc4207addf0)
	/go/src/github.com/cockroachdb/cockroach/pkg/util/stop/stopper.go:210 +0x7d
created by github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).RunWorker
	/go/src/github.com/cockroachdb/cockroach/pkg/util/stop/stopper.go:211 +0x66

goroutine 39214 [select, 2 minutes]:
github.com/cockroachdb/cockroach/pkg/storage.(*baseQueue).processLoop.func1()
	/go/src/github.com/cockroachdb/cockroach/pkg/storage/queue.go:490 +0x39e
github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).RunWorker.func1(0xc421324000, 0xc42015e0c0)
	/go/src/github.com/cockroachdb/cockroach/pkg/util/stop/stopper.go:210 +0x7d
created by github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).RunWorker
	/go/src/github.com/cockroachdb/cockroach/pkg/util/stop/stopper.go:211 +0x66

goroutine 39054 [semacquire]:
sync.runtime_notifyListWait(0xc4204be610, 0xc400000569)
	/usr/local/go/src/runtime/sema.go:297 +0x10b
sync.(*Cond).Wait(0xc4204be600)
	/usr/local/go/src/sync/cond.go:57 +0x89
github.com/cockroachdb/cockroach/pkg/storage.(*raftScheduler).worker(0xc420370e10, 0xc421324000)
	/go/src/github.com/cockroachdb/cockroach/pkg/storage/scheduler.go:196 +0xac
github.com/cockroachdb/cockroach/pkg/storage.(*raftScheduler).Start.func2()
	/go/src/github.com/cockroachdb/cockroach/pkg/storage/scheduler.go:165 +0x33
github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).RunWorker.func1(0xc421324000, 0xc4213feee0)
	/go/src/github.com/cockroachdb/cockroach/pkg/util/stop/stopper.go:210 +0x7d
created by github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).RunWorker
	/go/src/github.com/cockroachdb/cockroach/pkg/util/stop/stopper.go:211 +0x66

goroutine 39376 [select]:
github.com/cockroachdb/cockroach/pkg/storage.(*Store).startGossip.func4()
	/go/src/github.com/cockroachdb/cockroach/pkg/storage/store.go:1428 +0x51d
github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).RunWorker.func1(0xc421324000, 0xc4208d0fa0)
	/go/src/github.com/cockroachdb/cockroach/pkg/util/stop/stopper.go:210 +0x7d
created by github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).RunWorker
	/go/src/github.com/cockroachdb/cockroach/pkg/util/stop/stopper.go:211 +0x66

goroutine 39211 [select, 2 minutes]:
github.com/cockroachdb/cockroach/pkg/storage.(*baseQueue).processLoop.func1()
	/go/src/github.com/cockroachdb/cockroach/pkg/storage/queue.go:490 +0x39e
github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).RunWorker.func1(0xc421324000, 0xc42015e040)
	/go/src/github.com/cockroachdb/cockroach/pkg/util/stop/stopper.go:210 +0x7d
created by github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).RunWorker
	/go/src/github.com/cockroachdb/cockroach/pkg/util/stop/stopper.go:211 +0x66

goroutine 39056 [semacquire]:
sync.runtime_notifyListWait(0xc4204be610, 0xc400000571)
	/usr/local/go/src/runtime/sema.go:297 +0x10b
sync.(*Cond).Wait(0xc4204be600)
	/usr/local/go/src/sync/cond.go:57 +0x89
github.com/cockroachdb/cockroach/pkg/storage.(*raftScheduler).worker(0xc420370e10, 0xc421324000)
	/go/src/github.com/cockroachdb/cockroach/pkg/storage/scheduler.go:196 +0xac
github.com/cockroachdb/cockroach/pkg/storage.(*raftScheduler).Start.func2()
	/go/src/github.com/cockroachdb/cockroach/pkg/storage/scheduler.go:165 +0x33
github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).RunWorker.func1(0xc421324000, 0xc4213fef20)
	/go/src/github.com/cockroachdb/cockroach/pkg/util/stop/stopper.go:210 +0x7d
created by github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).RunWorker
	/go/src/github.com/cockroachdb/cockroach/pkg/util/stop/stopper.go:211 +0x66

goroutine 39045 [chan receive, 2 minutes]:
github.com/cockroachdb/cockroach/pkg/storage_test.TestStoreRangeGossipOnSplits(0xc4210d61a0)
	/go/src/github.com/cockroachdb/cockroach/pkg/storage/client_split_test.go:1799 +0x2f1
testing.tRunner(0xc4210d61a0, 0x1d91248)
	/usr/local/go/src/testing/testing.go:657 +0x96
created by testing.(*T).Run
	/usr/local/go/src/testing/testing.go:697 +0x2ca

goroutine 39365 [semacquire]:
sync.runtime_notifyListWait(0xc4204be610, 0xc400000570)
	/usr/local/go/src/runtime/sema.go:297 +0x10b
sync.(*Cond).Wait(0xc4204be600)
	/usr/local/go/src/sync/cond.go:57 +0x89
github.com/cockroachdb/cockroach/pkg/storage.(*raftScheduler).worker(0xc420370e10, 0xc421324000)
	/go/src/github.com/cockroachdb/cockroach/pkg/storage/scheduler.go:196 +0xac
github.com/cockroachdb/cockroach/pkg/storage.(*raftScheduler).Start.func2()
	/go/src/github.com/cockroachdb/cockroach/pkg/storage/scheduler.go:165 +0x33
github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).RunWorker.func1(0xc421324000, 0xc4213fefc0)
	/go/src/github.com/cockroachdb/cockroach/pkg/util/stop/stopper.go:210 +0x7d
created by github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).RunWorker
	/go/src/github.com/cockroachdb/cockroach/pkg/util/stop/stopper.go:211 +0x66

goroutine 39370 [semacquire]:
sync.runtime_notifyListWait(0xc4204be610, 0xc40000056c)
	/usr/local/go/src/runtime/sema.go:297 +0x10b
sync.(*Cond).Wait(0xc4204be600)
	/usr/local/go/src/sync/cond.go:57 +0x89
github.com/cockroachdb/cockroach/pkg/storage.(*raftScheduler).worker(0xc420370e10, 0xc421324000)
	/go/src/github.com/cockroachdb/cockroach/pkg/storage/scheduler.go:196 +0xac
github.com/cockroachdb/cockroach/pkg/storage.(*raftScheduler).Start.func2()
	/go/src/github.com/cockroachdb/cockroach/pkg/storage/scheduler.go:165 +0x33
github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).RunWorker.func1(0xc421324000, 0xc4213ff060)
	/go/src/github.com/cockroachdb/cockroach/pkg/util/stop/stopper.go:210 +0x7d
created by github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).RunWorker
	/go/src/github.com/cockroachdb/cockroach/pkg/util/stop/stopper.go:211 +0x66

goroutine 39049 [select]:
github.com/cockroachdb/cockroach/pkg/kv.(*TxnCoordSender).printStatsLoop(0xc420f9a420, 0x2b6cf28a50c8, 0xc420014e28)
	/go/src/github.com/cockroachdb/cockroach/pkg/kv/txn_coord_sender.go:207 +0xb4a
github.com/cockroachdb/cockroach/pkg/kv.NewTxnCoordSender.func1()
	/go/src/github.com/cockroachdb/cockroach/pkg/kv/txn_coord_sender.go:191 +0x98
github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).RunWorker.func1(0xc421324000, 0xc4207ac7e0)
	/go/src/github.com/cockroachdb/cockroach/pkg/util/stop/stopper.go:210 +0x7d
created by github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).RunWorker
	/go/src/github.com/cockroachdb/cockroach/pkg/util/stop/stopper.go:211 +0x66

goroutine 39050 [chan receive, 2 minutes]:
github.com/cockroachdb/cockroach/pkg/storage.(*raftScheduler).Start.func1()
	/go/src/github.com/cockroachdb/cockroach/pkg/storage/scheduler.go:155 +0x55
github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).RunWorker.func1(0xc421324000, 0xc4213fee60)
	/go/src/github.com/cockroachdb/cockroach/pkg/util/stop/stopper.go:210 +0x7d
created by github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).RunWorker
	/go/src/github.com/cockroachdb/cockroach/pkg/util/stop/stopper.go:211 +0x66

goroutine 39373 [select]:
github.com/cockroachdb/cockroach/pkg/storage.(*Store).startCoalescedHeartbeatsLoop.func1()
	/go/src/github.com/cockroachdb/cockroach/pkg/storage/store.go:3530 +0x167
github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).RunWorker.func1(0xc421324000, 0xc4207ade00)
	/go/src/github.com/cockroachdb/cockroach/pkg/util/stop/stopper.go:210 +0x7d
created by github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).RunWorker
	/go/src/github.com/cockroachdb/cockroach/pkg/util/stop/stopper.go:211 +0x66

goroutine 39057 [semacquire]:
sync.runtime_notifyListWait(0xc4204be610, 0xc400000572)
	/usr/local/go/src/runtime/sema.go:297 +0x10b
sync.(*Cond).Wait(0xc4204be600)
	/usr/local/go/src/sync/cond.go:57 +0x89
github.com/cockroachdb/cockroach/pkg/storage.(*raftScheduler).worker(0xc420370e10, 0xc421324000)
	/go/src/github.com/cockroachdb/cockroach/pkg/storage/scheduler.go:196 +0xac
github.com/cockroachdb/cockroach/pkg/storage.(*raftScheduler).Start.func2()
	/go/src/github.com/cockroachdb/cockroach/pkg/storage/scheduler.go:165 +0x33
github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).RunWorker.func1(0xc421324000, 0xc4213fef40)
	/go/src/github.com/cockroachdb/cockroach/pkg/util/stop/stopper.go:210 +0x7d
created by github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).RunWorker
	/go/src/github.com/cockroachdb/cockroach/pkg/util/stop/stopper.go:211 +0x66

goroutine 39213 [select, 2 minutes]:
github.com/cockroachdb/cockroach/pkg/storage.(*baseQueue).processLoop.func1()
	/go/src/github.com/cockroachdb/cockroach/pkg/storage/queue.go:490 +0x39e
github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).RunWorker.func1(0xc421324000, 0xc42015e080)
	/go/src/github.com/cockroachdb/cockroach/pkg/util/stop/stopper.go:210 +0x7d
created by github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).RunWorker
	/go/src/github.com/cockroachdb/cockroach/pkg/util/stop/stopper.go:211 +0x66

goroutine 39378 [select]:
github.com/cockroachdb/cockroach/pkg/storage.(*Store).startGossip.func4()
	/go/src/github.com/cockroachdb/cockroach/pkg/storage/store.go:1428 +0x51d
github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).RunWorker.func1(0xc421324000, 0xc4208d1040)
	/go/src/github.com/cockroachdb/cockroach/pkg/util/stop/stopper.go:210 +0x7d
created by github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).RunWorker
	/go/src/github.com/cockroachdb/cockroach/pkg/util/stop/stopper.go:211 +0x66

goroutine 39215 [select, 2 minutes]:
github.com/cockroachdb/cockroach/pkg/storage.(*baseQueue).processLoop.func1()
	/go/src/github.com/cockroachdb/cockroach/pkg/storage/queue.go:490 +0x39e
github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).RunWorker.func1(0xc421324000, 0xc42034a000)
	/go/src/github.com/cockroachdb/cockroach/pkg/util/stop/stopper.go:210 +0x7d
created by github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).RunWorker
	/go/src/github.com/cockroachdb/cockroach/pkg/util/stop/stopper.go:211 +0x66

goroutine 39047 [chan receive, 2 minutes]:
github.com/cockroachdb/cockroach/pkg/rpc.NewContext.func1()
	/go/src/github.com/cockroachdb/cockroach/pkg/rpc/context.go:170 +0x76
github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).RunWorker.func1(0xc421324000, 0xc4213fe140)
	/go/src/github.com/cockroachdb/cockroach/pkg/util/stop/stopper.go:210 +0x7d
created by github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).RunWorker
	/go/src/github.com/cockroachdb/cockroach/pkg/util/stop/stopper.go:211 +0x66

goroutine 39367 [semacquire]:
sync.runtime_notifyListWait(0xc4204be610, 0xc40000056f)
	/usr/local/go/src/runtime/sema.go:297 +0x10b
sync.(*Cond).Wait(0xc4204be600)
	/usr/local/go/src/sync/cond.go:57 +0x89
github.com/cockroachdb/cockroach/pkg/storage.(*raftScheduler).worker(0xc420370e10, 0xc421324000)
	/go/src/github.com/cockroachdb/cockroach/pkg/storage/scheduler.go:196 +0xac
github.com/cockroachdb/cockroach/pkg/storage.(*raftScheduler).Start.func2()
	/go/src/github.com/cockroachdb/cockroach/pkg/storage/scheduler.go:165 +0x33
github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).RunWorker.func1(0xc421324000, 0xc4213ff000)
	/go/src/github.com/cockroachdb/cockroach/pkg/util/stop/stopper.go:210 +0x7d
created by github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).RunWorker
	/go/src/github.com/cockroachdb/cockroach/pkg/util/stop/stopper.go:211 +0x66

goroutine 39377 [select]:
github.com/cockroachdb/cockroach/pkg/storage.(*Store).startGossip.func4()
	/go/src/github.com/cockroachdb/cockroach/pkg/storage/store.go:1428 +0x51d
github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).RunWorker.func1(0xc421324000, 0xc4208d0ff0)
	/go/src/github.com/cockroachdb/cockroach/pkg/util/stop/stopper.go:210 +0x7d
created by github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).RunWorker
	/go/src/github.com/cockroachdb/cockroach/pkg/util/stop/stopper.go:211 +0x66

goroutine 39363 [semacquire]:
sync.runtime_notifyListWait(0xc4204be610, 0xc400000574)
	/usr/local/go/src/runtime/sema.go:297 +0x10b
sync.(*Cond).Wait(0xc4204be600)
	/usr/local/go/src/sync/cond.go:57 +0x89
github.com/cockroachdb/cockroach/pkg/storage.(*raftScheduler).worker(0xc420370e10, 0xc421324000)
	/go/src/github.com/cockroachdb/cockroach/pkg/storage/scheduler.go:196 +0xac
github.com/cockroachdb/cockroach/pkg/storage.(*raftScheduler).Start.func2()
	/go/src/github.com/cockroachdb/cockroach/pkg/storage/scheduler.go:165 +0x33
github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).RunWorker.func1(0xc421324000, 0xc4213fef80)
	/go/src/github.com/cockroachdb/cockroach/pkg/util/stop/stopper.go:210 +0x7d
created by github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).RunWorker
	/go/src/github.com/cockroachdb/cockroach/pkg/util/stop/stopper.go:211 +0x66

goroutine 39368 [semacquire]:
sync.runtime_notifyListWait(0xc4204be610, 0xc40000056b)
	/usr/local/go/src/runtime/sema.go:297 +0x10b
sync.(*Cond).Wait(0xc4204be600)
	/usr/local/go/src/sync/cond.go:57 +0x89
github.com/cockroachdb/cockroach/pkg/storage.(*raftScheduler).worker(0xc420370e10, 0xc421324000)
	/go/src/github.com/cockroachdb/cockroach/pkg/storage/scheduler.go:196 +0xac
github.com/cockroachdb/cockroach/pkg/storage.(*raftScheduler).Start.func2()
	/go/src/github.com/cockroachdb/cockroach/pkg/storage/scheduler.go:165 +0x33
github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).RunWorker.func1(0xc421324000, 0xc4213ff020)
	/go/src/github.com/cockroachdb/cockroach/pkg/util/stop/stopper.go:210 +0x7d
created by github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).RunWorker
	/go/src/github.com/cockroachdb/cockroach/pkg/util/stop/stopper.go:211 +0x66

goroutine 39210 [select, 2 minutes]:
github.com/cockroachdb/cockroach/pkg/storage.(*baseQueue).processLoop.func1()
	/go/src/github.com/cockroachdb/cockroach/pkg/storage/queue.go:490 +0x39e
github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).RunWorker.func1(0xc421324000, 0xc42015e020)
	/go/src/github.com/cockroachdb/cockroach/pkg/util/stop/stopper.go:210 +0x7d
created by github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).RunWorker
	/go/src/github.com/cockroachdb/cockroach/pkg/util/stop/stopper.go:211 +0x66

goroutine 39362 [semacquire]:
sync.runtime_notifyListWait(0xc4204be610, 0xc400000573)
	/usr/local/go/src/runtime/sema.go:297 +0x10b
sync.(*Cond).Wait(0xc4204be600)
	/usr/local/go/src/sync/cond.go:57 +0x89
github.com/cockroachdb/cockroach/pkg/storage.(*raftScheduler).worker(0xc420370e10, 0xc421324000)
	/go/src/github.com/cockroachdb/cockroach/pkg/storage/scheduler.go:196 +0xac
github.com/cockroachdb/cockroach/pkg/storage.(*raftScheduler).Start.func2()
	/go/src/github.com/cockroachdb/cockroach/pkg/storage/scheduler.go:165 +0x33
github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).RunWorker.func1(0xc421324000, 0xc4213fef60)
	/go/src/github.com/cockroachdb/cockroach/pkg/util/stop/stopper.go:210 +0x7d
created by github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).RunWorker
	/go/src/github.com/cockroachdb/cockroach/pkg/util/stop/stopper.go:211 +0x66

goroutine 39369 [semacquire]:
sync.runtime_notifyListWait(0xc4204be610, 0xc40000056e)
	/usr/local/go/src/runtime/sema.go:297 +0x10b
sync.(*Cond).Wait(0xc4204be600)
	/usr/local/go/src/sync/cond.go:57 +0x89
github.com/cockroachdb/cockroach/pkg/storage.(*raftScheduler).worker(0xc420370e10, 0xc421324000)
	/go/src/github.com/cockroachdb/cockroach/pkg/storage/scheduler.go:196 +0xac
github.com/cockroachdb/cockroach/pkg/storage.(*raftScheduler).Start.func2()
	/go/src/github.com/cockroachdb/cockroach/pkg/storage/scheduler.go:165 +0x33
github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).RunWorker.func1(0xc421324000, 0xc4213ff040)
	/go/src/github.com/cockroachdb/cockroach/pkg/util/stop/stopper.go:210 +0x7d
created by github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).RunWorker
	/go/src/github.com/cockroachdb/cockroach/pkg/util/stop/stopper.go:211 +0x66

goroutine 39216 [select, 2 minutes]:
github.com/cockroachdb/cockroach/pkg/storage.(*replicaScanner).waitEnabled(0xc42036f5c0, 0xc421324000, 0xc42036f600)
	/go/src/github.com/cockroachdb/cockroach/pkg/storage/scanner.go:307 +0x1ec
github.com/cockroachdb/cockroach/pkg/storage.(*replicaScanner).scanLoop.func1()
	/go/src/github.com/cockroachdb/cockroach/pkg/storage/scanner.go:260 +0x600
github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).RunWorker.func1(0xc421324000, 0xc42034a060)
	/go/src/github.com/cockroachdb/cockroach/pkg/util/stop/stopper.go:210 +0x7d
created by github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).RunWorker
	/go/src/github.com/cockroachdb/cockroach/pkg/util/stop/stopper.go:211 +0x66
--- FAIL: testrace (proposer evaluated kv)/TestLogic (310.940s)
null
--- FAIL: testrace (proposer evaluated kv)/TestLogic/distsql (157.740s)
null
--- FAIL: testrace (proposer evaluated kv)/TestLogic/distsql/pipelining (1.570s)
testing.go:610: race detected during execution of test
------- Stdout: -------
test logs captured to: /go/src/github.com/cockroachdb/cockroach/artifacts/TestLogic799627757  (use -show-logs to present inline)
--- new user: root
--- queries start here
--- done: testdata/logic_test/pipelining: 86 tests, 0 failures
--- total progress: 6332 statements/queries

```
Please assign, take a look and update the issue accordingly.

--- a/pkg/internal/client/db_test.go
+++ b/pkg/internal/client/db_test.go
@@ -402,6 +402,7 @@ func TestCommonMethods(t *testing.T) {
 		{txnType, "OrigTimestamp"}:             {},
 		{txnType, "Proto"}:                     {},
 		{txnType, "UserPriority"}:              {},
+		{txnType, "AnchorKey"}:                 {},
 	}
 
 	for b := range omittedChecks {

--- a/pkg/internal/client/txn.go
+++ b/pkg/internal/client/txn.go
@@ -199,6 +199,14 @@ func (txn *Txn) OrigTimestamp() hlc.Timestamp {
 	return txn.mu.Proto.OrigTimestamp
 }
 
+// AnchorKey returns the transaction's anchor key. The caller should treat the
+// returned byte slice as immutable.
+func (txn *Txn) AnchorKey() []byte {
+	txn.mu.Lock()
+	defer txn.mu.Unlock()
+	return txn.mu.Proto.Key
+}
+
 // SetTxnAnchorKey sets the key at which to anchor the transaction record. The
 // transaction anchor key defaults to the first key written in a transaction.
 func (txn *Txn) SetTxnAnchorKey(key roachpb.Key) error {

--- a/pkg/sql/executor.go
+++ b/pkg/sql/executor.go
@@ -1369,7 +1369,7 @@ func (e *Executor) shouldUseDistSQL(planner *planner, plan planNode) (bool, erro
 	// Temporary workaround for #13376: if the transaction modified something,
 	// TxnCoordSender will freak out if it sees scans in this txn from other
 	// nodes. We detect this by checking if the transaction's "anchor" key is set.
-	if planner.txn.Proto().TxnMeta.Key != nil {
+	if planner.txn.AnchorKey() != nil {
 		err = errors.New("writing txn")
 	} else {
 		// Trigger limit propagation.

